spanner: return type of toListAsync changed

```
[ERROR] /tmpfs/src/github/java-docs-samples/spanner/cloud-client/src/main/java/com/example/spanner/AsyncQueryToListAsyncExample.java:[72,31] error: incompatible types: no instance(s) of type variable(s) T exist so that ApiFuture<List<T>> conforms to ApiFuture<ImmutableList<Album>>
  where T is a type-variable:
    T extends Object declared in method <T>toListAsync(Function<StructReader,T>,Executor)
```

https://github.com/GoogleCloudPlatform/java-docs-samples/pull/3823

--- a/spanner/cloud-client/src/main/java/com/example/spanner/AsyncQueryToListAsyncExample.java
+++ b/spanner/cloud-client/src/main/java/com/example/spanner/AsyncQueryToListAsyncExample.java
@@ -24,7 +24,7 @@ import com.google.cloud.spanner.DatabaseId;
 import com.google.cloud.spanner.Spanner;
 import com.google.cloud.spanner.SpannerOptions;
 import com.google.cloud.spanner.Statement;
-import com.google.common.collect.ImmutableList;
+import java.util.List;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
@@ -62,7 +62,7 @@ class AsyncQueryToListAsyncExample {
   static void asyncQueryToList(DatabaseClient client)
       throws InterruptedException, ExecutionException, TimeoutException {
     ExecutorService executor = Executors.newSingleThreadExecutor();
-    ApiFuture<ImmutableList<Album>> albums;
+    ApiFuture<? extends List<Album>> albums;
     try (AsyncResultSet resultSet =
         client
             .singleUse()

